### ISSUE TITLE
Change RTI to handle new Interceptor class for static interop classes

### ISSUE THREAD

**Issue by srujzs** [dart-lang/sdk#47325]
Static interop classes will introduce a new type that is a supertype of all web `Native` classes and all interop classes. While the types that correspond to these classes will be erased at compile time, they still need to be made sure that all subtyping relationships act as expected.

**Comment by jodinathan** [comment]
does this new ```Native``` supertype change what can be done in JS interop?

**Comment by srujzs** [comment]
Yep. The goal is to allow `@JS` classes to be used on native types (which are currently reserved for `Native` classes) with the requirement that the classes need to be used statically (i.e. no instance members because of dynamic invocations) and is annotated with `@staticInterop`. For example,

```
@JS()
@staticInterop
class Window {}

extension on Window {
  external dynamic get navigator;
}
```

can be used to interop with `Window` objects. The pieces are there, there's just a dependency that needs to be updated before this is shipped.
